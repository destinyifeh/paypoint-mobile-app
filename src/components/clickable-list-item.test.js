'use strict';

import React from 'react';
import { configure, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

import ClickableListItem from './clickable-list-item';
import { TouchableOpacity } from 'react-native';

configure({
  adapter: new Adapter()
});

it('renders correctly', () => {
  const tree = renderer
    .create(<ClickableListItem />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('onPressOut functions as expected', () => {
  const onPressEvent = jest.fn();
  onPressEvent.mockReturnValue('Link on press invoked');
  const wrapper = shallow(<ClickableListItem
    onPressOut={ onPressEvent } 
  />);
  wrapper.find(TouchableOpacity).first().props().onPress();
  expect(onPressEvent.mock.calls.length).toBe(1);
})

it('onPress functions as expected', () => {
  const onPressEvent = jest.fn();
  onPressEvent.mockReturnValue('Link on press invoked');
  const wrapper = shallow(<ClickableListItem
    onPress={ onPressEvent } 
  />);
  wrapper.find(TouchableOpacity).first().props().onPress();
  expect(onPressEvent.mock.calls.length).toBe(1);
})

it('TouchableOpacity onPressOut has no effect', () => {
  const onPressEvent = jest.fn();
  onPressEvent.mockReturnValue('Link on press invoked');
  const wrapper = shallow(<ClickableListItem
    onPress={ onPressEvent } 
  />);
  wrapper.find(TouchableOpacity).first().props().onPressOut();
  expect(onPressEvent.mock.calls.length).toBe(0);
})
