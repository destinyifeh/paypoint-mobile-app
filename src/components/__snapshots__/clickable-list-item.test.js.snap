// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  clickable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
/>
`;
